src/main/java/org/broadinstitute/hellbender/engine/spark/GATKRegistrator.java

```diff
@@ -0,0 +1,27 @@
+package org.broadinstitute.hellbender.engine.spark;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.google.api.services.genomics.model.Read;
+import de.javakaffee.kryoserializers.UnmodifiableCollectionsSerializer;
+import org.apache.spark.serializer.KryoRegistrator;
+
+import java.util.Collections;
+
+/**
+ * GATKRegistrator registers Serializers for our project. We need a JsonSerializer for the Google Genomics classes
+ * and UnmodifiableCollectionsSerializer from a bug in the version of Kryo we're on.
+ */
+public class GATKRegistrator implements KryoRegistrator {
+    @SuppressWarnings("unchecked")
+    @Override
+    public void registerClasses(Kryo kryo) {
+
+        // JsonSerializer is needed for the Google Genomics classes like Read and Reference.
+        kryo.register(Read.class, new JsonSerializer<Read>());
+        // htsjdk.variant.variantcontext.CommonInfo has a Map<String, Object> that defaults to
+        // a Collections.unmodifiableMap. This can't be handled by the version of kryo used in Spark, it's fixed
+        // in newer versions, but we can't use those because of incompatibility with Spark. We just include the fix
+        // here.
+        kryo.register(Collections.unmodifiableMap(Collections.EMPTY_MAP).getClass(), new UnmodifiableCollectionsSerializer());
```

Comment:
Which version of Kryo is this fixed in? You've upgraded Kryo to 2.24.0, so presumably not that one. Can you add a comment saying why that version of Kryo is needed (and not the default one that Spark uses).